xarray/core/datatree.py

```diff
@@ -339,25 +373,27 @@ class DataTree(
     _children: dict[str, DataTree]
     _attrs: dict[Hashable, Any] | None
     _cache: dict[str, Any]
-    _coord_names: set[Hashable]
-    _dims: dict[Hashable, int]
+    _local_coord_names: set[Hashable]
+    _inherited_coord_names: set[Hashable]
+    _local_dims: dict[Hashable, int]
     _encoding: dict[Hashable, Any] | None
     _close: Callable[[], None] | None
-    _indexes: dict[Hashable, Index]
-    _variables: dict[Hashable, Variable]
+    _local_indexes: dict[Hashable, Index]
+    _local_variables: dict[Hashable, Variable]
```

Comment:
The only new internal attribute used here is `_inherited_coord_names`. Some of the others are renamed just to clarify that they only hold the information local to this node.